src/v/cluster/controller.cc

```diff
@@ -54,7 +54,7 @@ controller::controller(
   ss::sharded<partition_manager>& pm,
   ss::sharded<shard_table>& st,
   ss::sharded<storage::api>& storage,
-  storage::node_api& storage_node,
+  ss::sharded<storage::node_api>& storage_node,
```

Comment:
this commit could be squashed with the previous commit

Reply:
Yeah, I think so. Was cool to see the change on its own for my curiosity. I'll squash when I address other comments too.